### ISSUE TITLE
z_datetime:next_month/1 does not take into account leap years

### ISSUE THREAD

**Issue by Dirklectisch** [zotonic/zotonic#1596]
````erlang
> z_datetime:next_month({{2017,2,30},{15,25,28}}).      
{{2017,3,30},{15,25,28}}
````

The above code returns an invalid date.  This is a problem especially when you just want to know what the next month is and don't care about days at all. Formatting it with the template filter 'date' results in the following error:

````erlang
15:25:28.588 [warning] Date format on illegal date {{2017,2,30},{15,25,28}} (format "m"), error: if_clause
````

**Comment by Dirklectisch** [comment]
There is already a month_boundaries function that seems to honour leap years. I should probably just have the next_month function use that.

**Comment by mworrell** [comment]
I guess you meant to give `{{2017,1,30},{15,25,28}}` as an input (corrected the issue).

This code should be checking the leap years. Will check.

**Comment by mworrell** [comment]
Of course, there is a definition problem here.  What is the date if the day doesn't exist in the next month?

**Comment by mworrell** [comment]
Fixed, please try again with the master or 0.x

**Comment by Dirklectisch** [comment]
Excellent! Thanks @mworrell .
